### PR TITLE
Allow task runs without a flow run

### PR DESCRIPTION
**Pull request by zangell44**
This PR lays out the backend work necessary to allow for task runs without a flow run.

It does not include any client-side changes to make autonomous task runs a first class concept.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: src/prefect/server/models/task_runs.py [hunk 1]
```diff
@@ -51,34 +51,58 @@ async def create_task_run(
...
+            .limit(1)
+            .execution_options(populate_existing=True)
         )
-        .limit(1)
-        .execution_options(populate_existing=True)
-    )
-    result = await session.execute(query)
-    model = result.scalar()
+        result = await session.execute(query)
+        model = result.scalar()
```
Comment (desertaxle) [inline src/prefect/server/models/task_runs.py:105]: This code looks the same as 69-82. Can we consolidate?

**Review by chrisguidry — APPROVED** [review]
:exclamation:

File: src/prefect/server/database/migrations/versions/sqlite/2023_09_25_121806_8167af8df781_remove_flow_run_id_requirement_from_task_run.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table("task_run") as batch_op:
+        batch_op.alter_column("flow_run_id", nullable=True)
+
+    with op.batch_alter_table("log") as batch_op:
+        batch_op.alter_column("flow_run_id", nullable=True)
```
[the remark this answers is not in the document]
Comment (zangell44) [reply src/prefect/server/database/migrations/versions/sqlite/2023_09_25_121806_8167af8df781_remove_flow_run_id_requirement_from_task_run.py:22]: Foreign keys should be automatically disabled during sqlite migrations but I'll confirm

File: src/prefect/server/models/task_runs.py [hunk 3]
```diff
@@ -51,34 +51,55 @@ async def create_task_run(
...
```
Comment (urimandujano) [inline src/prefect/server/models/task_runs.py:102]: Q: I'm curious why we're doing an insert followed immediately by a query to get the model from the DB. Is that a form of sanity checking/retrieving values set by the DB? It looks like this is pre-existing behavior so 🤞🏼 the answer isn't lost to time.

**Review by urimandujano — APPROVED** [review]

**Pull request merged**